src/input/input_handler.cpp

```diff
@@ -0,0 +1,694 @@
+at once. Wheel inputs generate a timer that doesn't turn off their outputs automatically, but push a
+userevent to do so.
+
+What structs are needed?
+InputBinding(key1, key2, key3)
+ControllerOutput(button, axis) - we only need a const array of these, and one of the attr-s is
+always 0 BindingConnection(inputBinding (member), controllerOutput (ref to the array element))
+
+Things to always test before pushing like a dumbass:
+Button outputs
+Axis outputs
+Input hierarchy
+Multi key inputs
+Mouse to joystick
+Key toggle
+Joystick halfmode
+
+Don't be an idiot and test only the changed part expecting everything else to not be broken
+*/
+
+// Flags and values for varying purposes
+// todo: can we change these?
+int mouse_joystick_binding = 0;
+float mouse_deadzone_offset = 0.5, mouse_speed = 1, mouse_speed_offset = 0.1250;
+Uint32 mouse_polling_id = 0;
+bool mouse_enabled = false, leftjoystick_halfmode = false, rightjoystick_halfmode = false;
+
+std::list<std::pair<u32, bool>> pressed_keys;
+std::list<u32> toggled_keys;
+std::list<BindingConnection> connections = std::list<BindingConnection>();
+
+ControllerOutput output_array[] = {
+    // Important: these have to be the first, or else they will update in the wrong order
+    ControllerOutput((OrbisPadButtonDataOffset)LEFTJOYSTICK_HALFMODE),
+    ControllerOutput((OrbisPadButtonDataOffset)RIGHTJOYSTICK_HALFMODE),
+    ControllerOutput((OrbisPadButtonDataOffset)KEY_TOGGLE),
+
+    // Button mappings
+    ControllerOutput(OrbisPadButtonDataOffset::Triangle),
+    ControllerOutput(OrbisPadButtonDataOffset::Circle),
+    ControllerOutput(OrbisPadButtonDataOffset::Cross),
+    ControllerOutput(OrbisPadButtonDataOffset::Square),
+    ControllerOutput(OrbisPadButtonDataOffset::L1),
+    ControllerOutput(OrbisPadButtonDataOffset::L3),
+    ControllerOutput(OrbisPadButtonDataOffset::R1),
+    ControllerOutput(OrbisPadButtonDataOffset::R3),
+    ControllerOutput(OrbisPadButtonDataOffset::Options),
+    ControllerOutput(OrbisPadButtonDataOffset::TouchPad),
+    ControllerOutput(OrbisPadButtonDataOffset::Up),
+    ControllerOutput(OrbisPadButtonDataOffset::Down),
+    ControllerOutput(OrbisPadButtonDataOffset::Left),
+    ControllerOutput(OrbisPadButtonDataOffset::Right),
+
+    // Axis mappings
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::LeftX),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::LeftY),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::RightX),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::RightY),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::TriggerLeft),
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::TriggerRight),
+
+    // the "sorry, you gave an incorrect value, now we bind it to nothing" output
+    ControllerOutput(OrbisPadButtonDataOffset::None, Axis::AxisMax),
+};
+
+// We had to go through 3 files of indirection just to update a flag
+void ToggleMouseEnabled() {
+    mouse_enabled ^= true;
```

Comment:
```suggestion
    mouse_enabled = !mouse_enabled;
```
While this code is more optimal than the bitwise not, the compiler can already do this optimization and its less readable. So lets do the simpler thing methinks